src/main.py

```diff
@@ -44,6 +55,21 @@ async def on_close() -> None:
         await bc.finish(say_goodbye=False)
 
 
+@client.event
+async def on_message(message: Message) -> None:
+    if (
+        config.openai_api_key
+        and message.guild
+        and (
+            client.user in message.mentions
+            or any(role.name == client.user.name for role in message.role_mentions)
```

Comment:
```suggestion
            # For the case where a user accidentally mentions the bot's role
            # instead of their nick (which are typically named the same).
            or any(role.name == client.user.name for role in message.role_mentions)
```